std/process.d

```diff
@@ -1034,54 +1034,92 @@ private Pid spawnProcessPosix(scope const(char[])[] args,
 
             if (!(config.flags & Config.Flags.inheritFDs))
             {
-                // NOTE: malloc() and getrlimit() are not on the POSIX async
-                // signal safe functions list, but practically this should
-                // not be a problem. Java VM and CPython also use malloc()
-                // in its own implementation via opendir().
-                import core.stdc.stdlib : malloc;
-                import core.sys.posix.poll : pollfd, poll, POLLNVAL;
-                import core.sys.posix.sys.resource : rlimit, getrlimit, RLIMIT_NOFILE;
-
-                // Get the maximum number of file descriptors that could be open.
-                rlimit r;
-                if (getrlimit(RLIMIT_NOFILE, &r) != 0)
-                {
-                    abortOnError(forkPipeOut, InternalError.getrlimit, .errno);
-                }
-                immutable maxDescriptors = cast(int) r.rlim_cur;
+                import core.sys.posix.dirent : dirent, opendir, readdir, closedir, DIR;
+                import core.sys.posix.unistd : close;
+                import core.sys.posix.stdlib : atoi, malloc, free;
 
-                // The above, less stdin, stdout, and stderr
-                immutable maxToClose = maxDescriptors - 3;
+                // Missing druntime declaration
+                pragma(mangle, "dirfd")
+                extern(C) nothrow @nogc int dirfd(DIR* dir);
 
-                // Call poll() to see which ones are actually open:
-                auto pfds = cast(pollfd*) malloc(pollfd.sizeof * maxToClose);
-                if (pfds is null)
-                {
-                    abortOnError(forkPipeOut, InternalError.malloc, .errno);
-                }
-                foreach (i; 0 .. maxToClose)
+                // Try to open the directory /dev/fd or /proc/self/fd
+                DIR* dir = opendir("/dev/fd");
```

Comment:
This is Posix code, right? Does this exist for all posix platforms? BSDs, OSX, Cygwin, etc.